multitenancy/src/main/java/org/axonframework/extensions/multitenancy/components/deadletterqueue/MultiTenantDeadLetterQueue.java

```diff
@@ -0,0 +1,406 @@
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean process(Function<DeadLetter<? extends M>, EnqueueDecision<M>> processingTask) {
+        TenantDescriptor currentTenant = TenantWrappedTransactionManager.getCurrentTenant();
+        if (currentTenant != null) {
+            return fetchFromTenantSegment(currentTenant, seg -> seg.process(processingTask));
+        } else {
+            logger.info("No tenant found for current thread. Processing all tenants queues.");
+            return tenants.stream().map(tenant -> fetchFromTenantSegment(tenant, seg -> seg.process(processingTask)))
+                          .reduce(false, (a, b) -> a || b);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void clear() {
+        TenantDescriptor currentTenant = TenantWrappedTransactionManager.getCurrentTenant();
+        if (currentTenant != null) {
+            executeForTenantSegment(currentTenant, SequencedDeadLetterQueue::clear);
+        } else {
+            logger.info("No tenant found for current thread. Clearing all tenants queues.");
+            tenants.forEach(tenant -> executeForTenantSegment(tenant, SequencedDeadLetterQueue::clear));
+        }
+    }
+
+    private <R> R fetchFromTenantSegment(TenantDescriptor tenantDescriptor,
+                                         Function<SequencedDeadLetterQueue<M>, R> function) {
+        R res = new TenantWrappedTransactionManager(NoTransactionManager.INSTANCE,
+                                                  tenantDescriptor).fetchInTransaction(() -> function.apply(
+                getTenantSegment(tenantDescriptor)));
+        return res;
+    }
+
+    private void executeForTenantSegment(TenantDescriptor tenantDescriptor,
+                                         Consumer<SequencedDeadLetterQueue<M>> consumer) {
+        new TenantWrappedTransactionManager(NoTransactionManager.INSTANCE, tenantDescriptor).fetchInTransaction(() -> {
+            consumer.accept(getTenantSegment(tenantDescriptor));
+            return null;
+        });
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Registration registerTenant(TenantDescriptor tenantDescriptor) {
+        tenants.add(tenantDescriptor);
+        return () -> {
+            tenants.remove(tenantDescriptor);
+            tenantSegments.remove(tenantDescriptor);
+            return true;
+        };
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Registration registerAndStartTenant(TenantDescriptor tenantDescriptor) {
+        return registerTenant(tenantDescriptor);
+    }
+
+    /**
+     * Return processing group that this queue is bounded to.
+     *
+     * @return processing group that this queue is bounded to.
+     */
+    public String processingGroup() {
+        return processingGroup;
+    }
+
+    /**
+     * Builder class to instantiate a {@link MultiTenantDeadLetterQueue}.
```

Comment:
Both the `TargetTenantResolver` and `processingGroup` are hard requirements for this Builder. Would be good to describe that on the JavaDoc, so that users can benefit from that.